src/librustdoc/clean/types.rs

```diff
@@ -2186,6 +2199,20 @@ impl Impl {
             .map(|did| tcx.provided_trait_methods(did).map(|meth| meth.name).collect())
             .unwrap_or_default()
     }
+
+    crate fn provided_associated_items(&self, tcx: TyCtxt<'_>) -> FxHashSet<(Symbol, AssocKind)> {
```

Comment:
This mostly replaces `Impl::provided_trait_methods` (the method above) except for one usage in `conversions.rs` (JSON output). Should I change the JSON output to include all kinds of associated types?

Reply:
I think you should for coherency.